charts/elasticsearch/templates/_helpers.tpl

```diff
@@ -162,3 +162,19 @@ imagePullSecrets:
 {{- .Values.curator.age.timestring | squote}}
 {{- end -}}
 {{- end -}}
+
+{{- define "logging.indexNamePrefix" -}}
+{{- if .Values.global.loggingSidecar.enabled -}}
+{{- if .Values.global.logging.indexNamePrefix -}}
+{{- .Values.global.logging.indexNamePrefix  -}}
+{{- else -}}
+vector
+{{- end -}}
+{{- else -}}
+{{- if .Values.global.logging.indexNamePrefix -}}
+{{- .Values.global.logging.indexNamePrefix -}}
+{{- else -}}
+fluentd
+{{- end -}}
+{{- end -}}
+{{- end -}}
```

Comment:
This can probably be simplified with the same changes as the external-es-proxy suggestion.